scripts/west_commands/runners/teensy.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2024 NXP
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''Runner for teensy .'''
+
+import os
+from os import path
+
+from runners.core import ZephyrBinaryRunner, RunnerCaps
+
+class TeensyBinaryRunner(ZephyrBinaryRunner):
+    '''Runner front-end for teensy.'''
+
+    def __init__(self, cfg, mcu,
+                 teensy='teensy', erase=False):
+        super().__init__(cfg)
+
+        default = path.join(cfg.board_dir, 'support', 'teensy.yaml')
+        if path.exists(default):
+            self.teensyconfig = default
+        else:
+            self.teensyconfig = None
+
+        self.mcu_args = ['--mcu', mcu]
+        self.teensy = teensy
+        self.hex_name = cfg.hex_file
+
+    @classmethod
+    def name(cls):
+        return 'teensy'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'},
+                          dev_id=True, flash_addr=True, erase=True,
+                          tool_opt=True)
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument('--mcu', required=True,
+                            help='Teensy muc target')
+        parser.add_argument('--teensy', default='teensy_loader_cli',
+                            help='path to teensy cli tool, default is teensy_loader_cli')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        ret = TeensyBinaryRunner(
+            cfg, args.mcu,
+            teensy=args.teensy,
+            erase=args.erase)
+        return ret
+
+    def do_run(self, command, **kwargs):
+        self.require(self.teensy)
+        if command == 'flash':
+            self.flash(**kwargs)
+
+    def flash(self, **kwargs):
+        if self.hex_name is not None and os.path.isfile(self.hex_name):
+            fname = self.hex_name
+        else:
+            raise ValueError(
+                'Cannot flash; no hex ({}) files found. '.format(self.hex_name))
+
+        cmd = ([self.teensy] +
```

Comment:
I don't think using `-w` here with the `while True` loop here makes sense. My concern is that the loader will effectively freeze waiting for the device to appear. There are a few issues with this:
- if the image being flashed is large, the user may confuse the loader waiting with the image taking time to flash
- if the device is: not connected, not powered, or not in HalfKay bootloader mode there is no feedback to the user
- this same issue appears if the device is not accessible (IE if the udev rules don't give the user access to the teensy device)

Reply:
Thanks for the suggestions, I've removed the loop and -w option.
On success output is
```
[1/1] cd /home/peter/cognipilot/ws/zephyr/build/zephyr && ...e=/home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner teensy
-- runners.teensy: Flashing file: /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
-- runners.teensy: Success
```
On failure you get this
```
[1/1] cd /home/peter/cognipilot/ws/zephyr/build/zephyr && ...e=/home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.elf
-- west flash: using runner teensy
-- runners.teensy: Flashing file: /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
Unable to open device (hint: try -w option)

FATAL ERROR: command exited with status 1: teensy_loader_cli -v --mcu TEENSY41 /home/peter/cognipilot/ws/zephyr/build/zephyr/zephyr.hex
```